GVFS/GVFS.UnitTests/Git/GitAuthenticationTests.cs

```diff
@@ -168,17 +169,47 @@ public void TwoThreadsInterleavingFailuresShouldntStompASuccess()
             thread2Auth.ShouldEqual(thread1Auth, "The second thread stomp the first threads good auth string");
         }
 
+        [TestCase]
+        public void DontDoubleStoreExistingCredential()
+        {
+            MockTracer tracer = new MockTracer();
+            MockGitProcess gitProcess = this.GetGitProcess();
+
+            GitAuthentication dut = new GitAuthentication(gitProcess, "mock://repoUrl");
+            dut.TryInitializeAndRequireAuth(tracer, out _);
+
+            string authString;
+            dut.TryGetCredentials(tracer, out authString, out _).ShouldBeTrue();
+            dut.ConfirmCredentialsWorked(authString);
+            dut.ConfirmCredentialsWorked(authString);
+            dut.ConfirmCredentialsWorked(authString);
+            dut.ConfirmCredentialsWorked(authString);
+            dut.ConfirmCredentialsWorked(authString);
+
+            gitProcess.StoredCredentials.Count.ShouldEqual(1);
+            gitProcess.StoredCredentials.Single().Key.ShouldEqual("mock://repoUrl");
+        }
+
         private MockGitProcess GetGitProcess()
         {
             MockGitProcess gitProcess = new MockGitProcess();
             gitProcess.SetExpectedCommandResult("config gvfs.FunctionalTests.UserName", () => new GitProcess.Result(string.Empty, string.Empty, GitProcess.Result.GenericFailureCode));
             gitProcess.SetExpectedCommandResult("config gvfs.FunctionalTests.Password", () => new GitProcess.Result(string.Empty, string.Empty, GitProcess.Result.GenericFailureCode));
 
+            int approvals = 0;

```

Comment:
I don't see approvals (or revocations) used for validation.  What are the thoughts on this check?

Reply:
I did wonder this. I was wondering if really this `MockGitProcess` would be better being replaced with a `Moq.Mock` with calls to `Verify` to check the number of invocations of various methods.

I opted not to churn the code too much however. I would reconsider this decision however if there's support/consensus on making these unit tests more thorough?

Reply:
I'm not a big fan of unused code, but I can understand the argument of keeping things consistent.  It seems like if we haven't used this to verify yet, not sure we ever will.  I wouldn't block a PR over this just pointing out something odd.

Reply:
@jeschu1  I modified the `MockGitProcess` to track the number of approvals and rejections so we can assert the expected number of calls in the tests.